src/queue/job/v1/task_sync.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{
+    entity::*,
+    queue::job::{EnqueueJob, Job, JobError, SharedJobState, V1},
+};
+use sea_orm::{ColumnTrait, ConnectionTrait, EntityTrait, PaginatorTrait, QueryFilter};
+use serde::{Deserialize, Serialize};
+use time::Duration;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Copy)]
+pub struct TaskSync;
+
+const SYNC_PERIOD: Duration = Duration::weeks(1);
+
+impl TaskSync {
+    pub async fn perform(
+        &mut self,
+        job_state: &SharedJobState,
+        db: &impl ConnectionTrait,
+    ) -> Result<Option<EnqueueJob>, JobError> {
+        let aggregators = Aggregators::find()
+            .filter(AggregatorColumn::IsFirstParty.eq(true)) // eventually we may want to check for a capability
+            .all(db)
+            .await?;
+
+        for aggregator in aggregators {
+            let client = aggregator.client(job_state.http_client.clone());
+            for task_id in client.get_task_ids().await? {
+                if 0 == Tasks::find_by_id(&task_id).count(db).await? {
```

Comment:
we may eventually want to perform a single query to prefetch all ids for the given aggregator

Reply:
This likely also should constrain the query to a given aggregator, since it's just as invalid for the aggregator to have another aggregator's task as it is to have a task that we don't know about at all